pkg/reconciler/eventlistener/resources/service.go

```diff
@@ -99,6 +106,10 @@ func ServicePort(el *v1beta1.EventListener, c Config) corev1.ServicePort {
 			// We return port 8443 if TLS is enabled and the default HTTP port is set.
 			// This effectively makes 8443 the default HTTPS port unless a user explicitly sets a different port.
 			servicePortPort = 8443
+		} else {
+			// Return port 443 if TLS is enabled and the HTTP port is not the default one.
+			// This will allow traffic to be directed to the default HTTPS port.
+			servicePortPort = 443
 		}
```

Comment:
@waveywaves How about to remove 
```
if *c.Port == DefaultPort {
```
and just override `servicePortPort` to `8443` instead of confusing with `8443` and `443` 

WDYT?

Reply:
Could you explain why we need 443 as the default TLS port?

Reply:
After giving it some thought I realized that this may not be a great solution to the TLS port issue. Allowing port override should allow the user to configure the port as 443 if they want to. Having it as a default port is not a great idea fosho.